### PR TITLE
Use binary upload API

### PR DESCRIPTION
**Pull request by tsg**
This enables uploads above ~20MB.

### CODE REVIEW

File: app/api/images/route.ts [hunk 1]
```diff
@@ -37,15 +37,20 @@ export async function POST(request: Request) {
...
+  // Upload the file and attach it to the image record
+  await xata.files.upload(
+    {
+      table: 'image',
+      column: 'image',
+      record: record.id
+    },
+    fileData
+  );
+  await xata.db.image.update(record.id, { image: { mediaType: mimeType, name: fileName } });
```
Comment (SferaDev) [inline app/api/images/route.ts:52]: If we didn't care about the file name this call can be removed.

File: app/api/images/route.ts [hunk 2]
```diff
@@ -37,15 +37,20 @@ export async function POST(request: Request) {
...
   });
 
+  // Upload the file and attach it to the image record
+  await xata.files.upload(
+    {
+      table: 'image',
+      column: 'image',
+      record: record.id
+    },
+    fileData
```
Comment (SferaDev) [inline app/api/images/route.ts:50]: We are creating a Buffer from an ArrayBuffer some lines before, we could leave it as an array buffer and do

XataFile.fromArrayBuffer(data, { mediaType })

Comment (snide) [reply app/api/images/route.ts:50]: I tried this, but I don't think XataFile is meant to be used with binary upload. It resulted in a malformed item being uploaded. Maybe I got the syntax wrong? Sorry, the TS here in a bunch of these objects isn't very helpful.

Also, if we're supposed to be using `XataFile`, we should likely have a lot of this stuff in our docs. I haven't really been aware I should be using it at all.

image

File: app/api/images/route.ts (hunk 1 above)
Comment (snide) [reply app/api/images/route.ts]: Good point. I removed this and the related vars we were pushing to it.

**Review by snide — APPROVED** [review]
I think this works pretty good as is. Moving to binary upload is good to show so others can learn.

**Comment by tsg** [comment]
There might be an issue with the mime type. I just noticed that a file uploaded this way has the media type set to `application/json`, and another user sees the same thing.

I'm going to hold merging this, until we figure it out.

File: app/api/images/route.ts (hunk 2 above)
Comment (SferaDev) [reply app/api/images/route.ts:48]: You can do `xataFile.toBlob()`

**Comment by tsg** [comment]
Pushed another commit, that seems to work.

**Pull request closed without merging**